cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -274,31 +269,53 @@ impl ProcessStates {
             .map(|state| state.name.clone())
             .collect()
     }
+}
 
-    fn fmt_table(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        writeln!(f, "{:<20} {:<10} {:>8}", "NAME", "STATUS", "PID")?;
-        for proc in &self.0 {
-            writeln!(f, "{:<20} {:<10} {:>8}", proc.name, proc.status, proc.pid)?;
-        }
-        Ok(())
+#[derive(Clone, Debug, Serialize)]
+/// Simplified version of ProcessState for display in the CLI.
+pub struct ProcessStateDisplay {
+    name: String,
+    status: String,
+    pid: u64,
+}
+
+#[derive(Clone, Debug, Serialize)]
+/// Simplified version of ProcessStates for display in the CLI.
+pub struct ProcessStatesDisplay(Vec<ProcessStateDisplay>);
```

Comment:
**nit**: I'd debate that this should be defined along the `status` command
suggestion: comment before derives
suggestion: Description could describe that it renders a table

```suggestion
/// Display implementation for [ProcessStates] that renders processes as a table.
#[derive(Clone, Debug, Serialize)]
pub struct ProcessStatesDisplay(Vec<ProcessStateDisplay>);
```

Reply:
How do you feel about needing to make `ProcessState` public for the purpose of `generate_process_state` in the CLI tests?